shared/src/TwinCollectionArray.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Devices.Shared
+{
+    /// <summary>
+    /// Represents a property array in <see cref="TwinCollection"/>
+    /// </summary>
+    public class TwinCollectionArray : JArray
+    {
+        private const string MetadataName = "$metadata";
```

Comment:
Can you find a way to not have to duplicate these constants across files?

Reply:
sure, will do